Workaround to make --no-newline work

msabramo opened the pull request:
See: https://github.com/chalk/chalk-cli/pull/16#issuecomment-918490669

msabramo committed:
Workaround to make --no-newline work

Qix- commented:
Does line 64 need to be changed?

msabramo commented:
@Qix:

> Does line 64 need to be changed?

You're referring to this line:

```js
noNewline: {
```

?

Do you think this line should change? I'm not seeing a reason why, but maybe you're seeing something that I am not...?

msabramo committed:
Add test for --no-newline

msabramo commented:
I added a new test for the `--no-newline` case in a59ac22, so hopefully the tests make the case that everything is working as desired.

sindresorhus commented on cli.js:
```diff
@@ -83,7 +83,7 @@ function init(data) {
 	const fn = dotProp.get(chalk, styles.join('.'));
 	process.stdout.write(fn(data.replace(/\n$/, '')));
-	if (!cli.flags.noNewline) {
+	if (!cli.flags.noNewline && cli.flags.newline !== false) {
```
This could use a short code comment explaining the logic and linking to the `yarns-parser` issue.

msabramo replied:
Ah okay, but I didn't create a `yargs-parser` issue yet because I found a possible way to do it using an existing [`boolean-negation` feature in yargs-parser](https://github.com/yargs/yargs-parser#boolean-negation).

See https://github.com/chalk/chalk-cli/pull/16#issuecomment-918528286

That approach wouldn't require changing `yargs-parser` and would only require a small change to `meow`.

I thought that might be a pretty good approach, since it seemed to require only a small change to meow and since meow is one of your projects, it would be easy to get done.

Check out https://github.com/chalk/chalk-cli/pull/16#issuecomment-918528286 and see if you like that, or if you think it's better to request a feature from `yargs-parser`.

msabramo replied:
@sindresorhus: See 5e680e1. It adds a comment that explains and points the reader to https://github.com/chalk/chalk-cli/issues/30

We can discuss things in that issue and/or link to any meow or yargs-parser issues or PRs that are created.

sindresorhus replied:
I would prefer an option in `yarns-parser`. That would benefit all consumers using `yarns-parsers` and not just `meow`. It seems like it is something that would be generally useful.

msabramo replied:
Ok. Would you please copy this comment to https://github.com/chalk/chalk-cli/issues/30 ?

I think it will be nice to have it in an open issue so it is more likely to get eyeballs.

msabramo committed:
Add comment explaining complex line

and referencing issue created to track simplifying it:

https://github.com/chalk/chalk-cli/issues/30

Qix- commented:
@sindresorhus are we still waiting for feedback or is this good to go?

sindresorhus commented:
https://github.com/chalk/chalk-cli/pull/23/files#r708319532

msabramo commented:
🎉
